weave-js/src/components/PagePanelComponents/Home/Browse3/pages/PlaygroundPage/PlaygroundChat/LLMDropdown.tsx

```diff
@@ -57,21 +66,48 @@ export const LLMDropdown: React.FC<LLMDropdownProps> = ({
     refetchConfiguredProviders();
   }, [refetchConfiguredProviders]);
 
-  const isValueAvailable = llmDropdownOptions.find(
-    option =>
-      'llms' in option && option.llms?.some(llm => llm && llm.value === value)
-  );
+  const isValueAvailable = useMemo(() => {
+    return llmDropdownOptions.some(
+      (option: ProviderOption) =>
+        'llms' in option && option.llms?.some(llm => llm && llm.value === value)
+    );
+  }, [llmDropdownOptions, value]);
 
   useEffect(() => {
     if (!isValueAvailable && !areProvidersLoading) {
-      for (const option of llmDropdownOptions) {
-        for (const llm of option.llms) {
-          if (llm && llm.value && llm.max_tokens) {
-            onChange(llm.value, llm.max_tokens);
+      let firstAvailableLlm: LLMOption | null = null;
+
+      // Check if the value is a saved model
+      const savedModelOption = llmDropdownOptions.find(
+        option => option.value === 'saved-models'
```

Comment:
Is `"saved-models"` a special marker? Should it be saved as a constant somewhere?